### PR TITLE
gh-2828 Lock / unlock store

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #2828
Handles #2829

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: Multisig/Features/Security/ProtectedKeyStore.swift [hunk 1]
```diff
@@ -38,6 +44,35 @@ class ProtectedKeyStore: EncryptedStore {
         }
     }
 
+    func unlock(password userPassword: String? = nil) throws {
```
Comment (DmitryBespalov) [inline Multisig/Features/Security/ProtectedKeyStore.swift]: fyi, this has to be a derived password, not a plaintext user passcode

File: Multisig/Features/Security/SecurityCenter.swift [hunk 2]
```diff
@@ -20,6 +20,8 @@ class SecurityCenter {
     private static let appUnlockChallengeID = "global.safe.AppUnlockChallenge"
     private static let challenge = "I am not alive, but I grow; I don't have lungs, but I need air; I don't have a mouth, but water kills me. What am I?"
 
+    private var dataStoreKey: SecKey? = nil
```
Comment (DmitryBespalov) [inline Multisig/Features/Security/SecurityCenter.swift]: remnant, should probably go away

**Comment by biafra23** [comment]
If biometry is enabled when enabbling the security lock, then the unlock screen is show and the button has to be pressed to show the biometric prompt for the first time. According to the ticket this prompt should be shown to the user without pressing the button.

**Review by biafra23 — APPROVED** [review]
I approve this PR :-)

**Review by DmitryBespalov — APPROVED** [review]

**Pull request merged**